### ISSUE TITLE
Incorrect polynomial fitment

### ISSUE THREAD

**Issue by x-zvf** [gifnksm/polynomial-rs#25]
The degree 2 polynomial to   $f(0)=3849; f(1)=34331; f(2)=95175$$ is incorrectly fitted.

This library returns: $ 3848 + 15303 x + 15180 x^2 $
The correct result is: $3849 + 15301 x + 15181 x^2 $
This can be verified by res-subsituting the variables. In particular, the error in $f(0)$ is trivial

Wolframalpha gives [the correct result](https://www.wolframalpha.com/input?i=+degree+2+fit+for+%280%2C3849%29++%281%2C34331%29+%282%2C95175%29)

## Reproduction
`Cargo.tom`l

```toml
[dependencies]
polynomial = "0.2.6"
```

`src/main.rs`
```rust
use polynomial::Polynomial;
fn main() {
    let xs = vec![0, 1, 2];
    let ys = vec![3849, 34331, 95175];

    let poly = Polynomial::lagrange(&xs, &ys).unwrap();
    println!("Polynomial: {}", poly.pretty("x")); //3848+15303*x+15180*x^2
    // https://www.wolframalpha.com/input?i=+degree+2+fit+for+%280%2C3849%29++%281%2C34331%29+%282%2C95175%29
    // Mathematica gives the correct result: 3849 + 15301 x + 15181 x^2
    // error: -1 - 3x - 1x^2
}
```

**Comment by konsumlamm** [comment]
The reason this happens is that you're using integers, so you get rounding errors. Using floats, you get the correct result.

**Comment by x-zvf** [comment]
Fair enough, although from a user's perspective it does seem weird, that fitting purely integer points (with an integer result) yields rounding errors.
Since the only imporant property of  the Lagrange-polynomial is that it EXACTLY fits the points, maybe you should consider disallowing the usage of integers with the type system, or at least note this in the documentation
